internal/terraform/node_action_validate.go

```diff
@@ -82,16 +82,28 @@ func (n *NodeValidatableAction) Execute(ctx EvalContext, _ walkOperation) tfdiag
 		return diags
 	}
 
+	// If the action has no "config" attribute or an empty config (config = {}),
+	// we will send a fully typed null config. This appeases other functions (such as
+	// EvaluateBlock) that panic when they encounter an empty body.
 	var configVal cty.Value
 	var valDiags tfdiags.Diagnostics
-	if n.Config.Config != nil {
+	if n.Config.Config == nil || n.Config.Config == hcl.EmptyBody() {
+		configVal = cty.NullVal(n.Schema.ConfigSchema.ImpliedType())
+		if n.Schema.ConfigSchema.ContainsRequired() {
+			diags = diags.Append(&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Missing required configuration",
+				Detail:   fmt.Sprintf("The action %s has required configuration, but no config block was found.", n.Addr.String()),
+				Subject:  &n.Config.TypeRange,
+			})
+			return diags
+		}
```

Comment:
Just an idea, could we do:

```
configVal := cty.EmptyObjectVal
if err := convert.Convert(configVal, n.Schema.ConfigSchema.ImpliedType()); err != nil {
 // diag
}
```

I think that would then mimic the kind of behaviour you get with regular resources? Like, with resources if no attributes are specified then we don't send null we send an empty object and we'd get the same behaviour with this approach.

I think what you have works fine, but I wonder if the meaning of a missing config block should be implied to be same as an empty one (which is an empty object).

Reply:
Ahhhh thank you very much, greatly appreciated, that is what I was missing

Reply:
OH, huh, I was only seeing your first comment - I'm still going to play with your suggestion a bit and see if I can get closer to the correct handling, at least. I can (try to) manually populate the computed nulls if nothing else.